lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -321,17 +350,45 @@ def headers
         }
       end
 
+      def public_key_headers
+        {
+          'Authorization' => @options[:public_key],
+          'Content-Type' => 'application/json;charset=UTF-8'
+        }
+      end
+
+      def tokenize(payment_method, options = {})
+        post = {}
+        add_authorization_type(post, options)
+        add_payment_method(post, payment_method, options)
+        add_customer_data(post, options)
+        begin
+          response = parse(ssl_request(:post, url(post, :tokens, nil), post[:source].to_json, public_key_headers))
+          succeeded = success_from(:tokens, response)
+        rescue ResponseError => e
+          raise unless e.response.code.to_s =~ /4\d\d/
+
+          response = parse(e.response.body, error: e.response)
+        end
+        response(:tokens, succeeded, response)
+      end
+
       def url(_post, action, authorization)
-        if %i[authorize purchase credit].include?(action)
+        case action
+        when ->(a) { %i[authorize purchase credit].include?(a) }
```

Comment:
💬 Comment:

Small ruby trick => you achieve the same with the splat operator:

```ruby
      def url(_post, action, authorization)
        case action
        when *[:authorize, :purchase, :credit]
          "#{base_url}/payments"
        when *[:store, :unstore]
          "#{base_url}/instruments/#{authorization}"
        # ...
        # ...
        else
          "#{base_url}/payments/#{authorization}/#{action}"
        end
      end

```

Reply:
Nice